src/assembly/linux/main/README.md

```diff
@@ -32,6 +32,108 @@ Starting Linux demo... Use Ctrl+C to exit.
+
+```
+./openc3.sh start
+```
+
+After the containers start, open a web browser and navigate to `http://localhost:2900`. You will be prompted to configure a password, and then presented with the COSMOS interface. Some modules may still be loading, but after a minute the interface should be complete. Navigate to `ADMIN CONSOLE` and uninstall the "openc3-cosmos-demo" plugin by selecting the `Delete Plugin` icon.
+
+After installing and running COSMOS we need to build the Linux Example plugin configuration files. This will allow COSMOS to decode telemetry from the Linux assembly and properly format outgoing commands.
+
+These files are autocoded by Adamant. From the `adamant_example/src/assembly/linux/main` directory in your Adamant environment run:
+
+```
+$ redo cosmos_config
+```
+
+This generates the plugin configuration file and its respective command and telemetry configurations in the `adamant_example/src/assembly/linux/build/cosmos/plugin/` directory.
+
+OpenC3 COSMOS runs inside of its own Docker containers. The COSMOS example project provides the necessary tools and commands to create an interface plugin from our generated configuration files. Navigate to the `cosmos-project` directory and run:
+
+```
+$ ./openc3.sh cli generate plugin LINUX_EXAMPLE
+```
+
+This creates a new directory called `openc3-cosmos-linux-example`. Navigate to this directory and run:
+
+```
+$ ./../openc3.sh cli generate target LINUX_EXAMPLE
+```
+
+This creates the plugin directory structure as well as template configuration files that our generated configuration will replace. These files are located at:
+
+```
+cosmos-project/openc3-cosmos-linux-example/plugin.txt
+cosmos-project/openc3-cosmos-linux-example/targets/LINUX_EXAMPLE/cmd_tlm/cmd.txt
+cosmos-project/openc3-cosmos-linux-example/targets/LINUX_EXAMPLE/cmd_tlm/tlm.txt
+```
+
+Replace the contents of these text files with the contents of the corresponding files generated by Redo, located in the Adamant example project directory at:
+
+```
+adamant_example/src/assembly/linux/build/cosmos/plugin
+```
+
+The COSMOS plugin uses a default CRC protocol provided by COSMOS, but also needs a custom protocol to apply Adamant's command packet checksum correctly. The protocol file is located in the Adamant directory at:
+
+```
+adamant/gnd/cosmos/cmd_checksum.rb
+```
+
+And must be placed in the COSMOS plugin directory at:
+
+```
+cosmos-project/openc3-cosmos-linux-example/targets/LINUX_EXAMPLE/lib/cmd_checksum.rb
+```
+
+A helper script, which takes the Adamant assembly name and COSMOS install directory as arguments, is provided to copy the plugin configuration files, and any custom protocols requested by the configuration, to the correct directories. If the COSMOS and Adamant example project directories are adjacent, complete the configuration by navigating to `adamant_example/gnd/cosmos` on the host machine and running:
+
+```
+$ ./install_cosmos_plugin.sh linux_example cosmos-project
```

Comment:
I think using full paths here would be clearer, ie something like
```
$ adamant_example/gnd/cosmos/install_cosmos_plugin.sh adamant_example/src/assembly/linux_example /path/to/cosmos-project
```

Something like that.

Reply:
For this script the first argument is the name of the assembly, which is needed to complete parsing of the COSMOS directory structure, since it uses this name in different configurations. The second argument is a path, relative to the location of the `adamant_example` directory, should this be a full path instead of relative for the user?
Other instructions in the Readme direct the user to navigate to a directory before running a command or script, in this case should it be the full path after navigating to the top directory, e.g. `$ ./adamant_example/gnd/cosmos/install_cosmos_plugin.sh`?

Reply:
Once this script is fixed up, you should remove all the business about manually copying things above. We should just encourage users to use the script.

Reply:
I think the script should assume the directories are relative to `cwd`, ie. wherever the user is currently located, unless a full path is given. That is the way `cp` or `mv` works.

Reply:
Changed to use paths relative to CWD and update README to reflect this in [0b74e73](https://github.com/lasp/adamant_example/pull/18/commits/0b74e739e6f076d635e9c2c9f35a434ac0e2c7dd), plus supporting updates from https://github.com/Jbsco/adamant_example/pull/2 and https://github.com/Jbsco/adamant/pull/2.